main.cpp

```diff
@@ -0,0 +1,29 @@
+#include "Header.h"
+
+
+
+template <typename T, const unsigned int SIZE>
+double Point<T, SIZE>::distance(const Point & p) const {
+	double result;
+	for (int i = 0; i < SIZE; i++) {
+		result += std::pow(components_[i] - p.components_[i], 2)
+	}
+	return std::sqrt(result);
+}
```

Comment:
for class templates, the implementation of methods should be in the header file with everything else (except in certain situations, or if you do things a certain way... lots has been written on that, but easiest to just put it all in the header)